File: test/fixtures/ts-app/cart.test.js
import assert from 'node:assert/strict';
import { test } from 'node:test';
import { applyDiscount, formatMoney, subtotal } from './cart.js';

test('subtotal sums price times quantity', () => {
  assert.equal(subtotal([{ price: 250, qty: 2 }, { price: 100, qty: 1 }]), 600);
});

test('applyDiscount accepts a fraction', () => {
  assert.equal(applyDiscount(1000, 0.1), 900);
});

test('applyDiscount accepts a whole percentage', () => {
  // 25 means 25%, not 2500%. This is the seeded bug.
  assert.equal(applyDiscount(1000, 25), 750);
});

test('formatMoney renders cents as currency', () => {
  assert.equal(formatMoney(648), '$6.48');
});

File: test/fixtures/go-app/cart_test.go
package cart

import "testing"

func TestSubtotal(t *testing.T) {
	got := Subtotal([]Item{{Price: 250, Qty: 2}, {Price: 100, Qty: 1}})
	if got != 600 {
		t.Errorf("Subtotal() = %d, want 600", got)
	}
}

func TestApplyDiscountFraction(t *testing.T) {
	if got := ApplyDiscount(1000, 0.1); got != 900 {
		t.Errorf("ApplyDiscount(1000, 0.1) = %d, want 900", got)
	}
}

func TestApplyDiscountWholePercentage(t *testing.T) {
	// 25 means 25%, not 2500%. This is the seeded bug.
	if got := ApplyDiscount(1000, 25); got != 750 {
		t.Errorf("ApplyDiscount(1000, 25) = %d, want 750", got)
	}
}
